posts/2024-08-07-Project-goals.md

```diff
@@ -0,0 +1,89 @@
+---
+layout: post
+title: "Rust Project goals for 2024"
+author: Niko Matsakis
+team: Leadership Council <https://www.rust-lang.org/governance/teams/leadership-council>
+---
+
+With the merging of [RFC #3672][], the Rust project has selected a **slate of 26 Project Goals** for the second half of 2024 (2024H2). This is our first time running an [experimental new roadmapping process][RFC #3614]; we expect to be running the process roughly every six months. Of these goals, we have designated three of them as our **flagship goals**, representing our most ambitious and most impactful efforts: (1) finalize preparations for the Rust 2024 edition; (2) bring the Async Rust experience closer to parity with sync Rust; and (3) resolve the biggest blockers to Linux building on stable Rust. As the year progress we'll be posting regular updates on these 3 flagship goals along with the 23 others.
+
+[RFC #3672]: https://github.com/rust-lang/rfcs/pull/3672#issuecomment-2254599176
+[RFC #3614]: https://github.com/rust-lang/rfcs/pull/3614
+
+## Rust’s mission
+
+Ultimately, all the goals selected are intended to further Rust's mission of **empowering everyone to build reliable and efficient software**. Rust targets programs that prioritize
+
+* reliability and robustness;
+* performance, memory usage, and resource consumption; and
+* long-term maintenance and extensibility.
+
+We consider "any two out of the three" as the right heuristic for projects where Rust is a strong contender or possibly the best option.
+
+## Why these particular flagship goals?
+
+**2024 Edition.** 2024 will mark the 4th Rust edition, following on the 2015, 2018, and 2021 editions. Similar to the [2021 edition](https://rust-lang.github.io/rust-project-goals/2024h2/https://github.com/nikomatsakis/rfcs/blob/rfl-project-goal/text/3085-edition-2021.html), the 2024 edition is not a "major marketing push" but rather an opportunity to correct small ergonomic issues with Rust that will make it overall much easier to use. The changes planned for the 2024 edition will (1) support `-> impl Trait` and `async fn` in traits by aligning capture behavior; (2) permit (async) generators to be added in the future by reserving the `gen` keyword; and (3) alter fallback for the `!` type. The [plan] is to finalize development of 2024 features this year; the Edition itself is planned for Rust v1.85 (released to beta 2025-01-03 and to stable on 2025-02-20).
+
+**Async.** In 2024 we plan to deliver several critical async Rust building block features, most notably support for *async closures* and *`Send` bounds*. This is part of a multi-year program aiming to raise the experience of authoring "async Rust" to the same level of quality as "sync Rust". Async Rust is a crucial growth area, with 52% of the respondents in the [2023 Rust survey](https://blog.rust-lang.org/2024/02/19/2023-Rust-Annual-Survey-2023-results.html) indicating that they use Rust to build server-side or backend applications. 
```

Comment:
Is growth area really the right word? At least, it seems like the statistic isn't relevant - we would need a metric for people who don't use rust but want to or something like that?

Reply:
Changed to 'widely used', it doesn't matter that much. I do think it's a growth area but I agree that the stat alone doesn't show that. And in any case I think right now Rust isn't as hyper focused on growth *per se* as on doing a better job in the domains where we are successful.